pkg/protocol/faststate/machine_test.go

```diff
@@ -0,0 +1,232 @@
+package faststate
+
+import (
+	"context"
+	"encoding/json"
+	"math/rand"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/internal/testutils"
+	"github.com/keep-network/keep-core/pkg/net"
+	netLocal "github.com/keep-network/keep-core/pkg/net/local"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+)
+
+const (
+	partyCount        = 3
+	round2MessageType = "test/round_2_message"
+	round3MessageType = "test/round_3_message"
+)
+
+func TestExecute(t *testing.T) {
+	provider := netLocal.Connect()
+	channel, err := provider.BroadcastChannelFor("test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	channel.SetUnmarshaler(func() net.TaggedUnmarshaler {
+		return &round2Message{}
+	})
+	channel.SetUnmarshaler(func() net.TaggedUnmarshaler {
+		return &round3Message{}
+	})
+
+	ctx, cancelFn := context.WithCancel(context.Background())
```

Comment:
Can we rename it to:
```suggestion
	ctx, cancelCtx := context.WithCancel(context.Background())
```
? `cancelFn` does not reflect that this function cancels a context.

Reply:
14777d2f580d8259c457317cb7e4bd1c2b19e363